middleware.ts

```diff
@@ -0,0 +1,14 @@
+import { NextResponse } from "next/server";
+import type { NextRequest } from "next/server";
+
+export function middleware(request: NextRequest) {
+  // Check if the user is trying to access the root "/"
+  if (request.nextUrl.pathname === "/") {
+    return NextResponse.redirect(new URL("/login", request.url));
```

Comment:
Avoid unnecessary URL instantiation – Instead of creating a new URL object, use request.nextUrl.pathname directly.
Ensure middleware returns undefined if no redirection is needed – Explicitly return undefined when no redirection is required for clarity.
```suggestion
    return NextResponse.redirect("/login");
  }

  return NextResponse.next();
```

Reply:
modified